test/yaml-rest-runner/src/main/java/org/elasticsearch/test/rest/yaml/section/SkipCriteria.java

```diff
@@ -0,0 +1,13 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.test.rest.yaml.section;
+
+public interface SkipCriteria {
```

Comment:
Isn't this just a `Predicate<SkipSectionContext>`?

Reply:
Yes, that's true. I thought having an explicit interface was cleaner/nicer. I think it's weird to `implement Predicate<>`.
But I'm OK with `Predicate<SkipSectionContext>` + some sort of factory, e.g. `Predicate<SkipSectionContext> createVersionBasedSkipCriteria(args)` to replace the concrete classes and constructors. Wdyt?

Reply:
That's a good solution, it keeps the semantics clear whilst making it easy to create implementations

Reply:
I was initially thinking the same, but then `SkipCriteria` is a bit more self-explanatory than `Predicate<SkipSectionContext>`. But how about moving the two impls into the interface?

```
    static SkipCriteria skipOs(List<String> operatingSystems){
        return context -> operatingSystems.stream().anyMatch(context::clusterIsRunningOs);
    }
```